postreise/plot/plot_interconnection_map.py

```diff
@@ -1,249 +1,179 @@
-    branch_tx = branch_map.loc[branch_map.interconnect == "Texas"]
-    branch_mdc = grid.dcline
+    _check_grid(grid)
+    if not isinstance(branch_distance_cutoff, (int, float)):
+        raise TypeError("branch_distance_cutoff must be an int")
+    if branch_distance_cutoff <= 0:
+        raise ValueError("branch_distance_cutoff must be strictly positive")
+    if not isinstance(figsize, tuple):
+        raise TypeError("figsize must be a tuple")
+    if not (len(figsize) == 2 and all(e > 0 for e in figsize)):
+        raise ValueError("both elemets of figsize must be positive")
+    if not isinstance(branch_width_scale_factor, (int, float)):
+        raise TypeError("branch_width_scale_factor must be a int/float")
+    if branch_width_scale_factor < 0:
+        raise ValueError("branch_width_scale_factor must be positive")
+    if not isinstance(hvdc_width_scale_factor, (int, float)):
+        raise TypeError("hvdc_width_scale_factor must be a int/float")
+    if hvdc_width_scale_factor < 0:
+        raise ValueError("hvdc_width_scale_factor must be positive")
+    if not isinstance(b2b_size_scale_factor, (int, float)):
+        raise TypeError("b2b_size_scale_factor must be a int/float")
+    if b2b_size_scale_factor < 0:
+        raise ValueError("b2b_size_scale_factor must be positive")
+    if not isinstance(background_map, bool):
+        raise TypeError("background_map must be a bool")
 
-    branch_mdc["from_lon"] = branch_bus.loc[branch_mdc.from_bus_id, "lon"].values
-    branch_mdc["from_lat"] = branch_bus.loc[branch_mdc.from_bus_id, "lat"].values
-    branch_mdc["to_lon"] = branch_bus.loc[branch_mdc.to_bus_id, "lon"].values
-    branch_mdc["to_lat"] = branch_bus.loc[branch_mdc.to_bus_id, "lat"].values
-    branch_mdc = project_branch(branch_mdc)
-
-    # back to backs are index 0-8, treat separately
-    branch_mdc1 = branch_mdc.iloc[
+    # branches
+    branch = grid.branch.copy()
+    branch["to_coord"] = list(zip(branch["to_lat"], branch["to_lon"]))
+    branch["from_coord"] = list(zip(branch["from_lat"], branch["from_lon"]))
+    branch["dist"] = branch.apply(
+        lambda row: distance.haversine(row["to_coord"], row["from_coord"]), axis=1
+    )
+    branch = branch.loc[branch["dist"] > branch_distance_cutoff]
+    branch = project_branch(branch)
+
+    branch_west = branch.loc[branch["interconnect"] == "Western"]
+    branch_east = branch.loc[branch["interconnect"] == "Eastern"]
+    branch_tx = branch.loc[branch["interconnect"] == "Texas"]
+
+    # HVDC lines
+    all_dcline = grid.dcline.copy()
+    all_dcline["from_lon"] = grid.bus.loc[all_dcline["from_bus_id"], "lon"].values
+    all_dcline["from_lat"] = grid.bus.loc[all_dcline["from_bus_id"], "lat"].values
+    all_dcline["to_lon"] = grid.bus.loc[all_dcline["to_bus_id"], "lon"].values
+    all_dcline["to_lat"] = grid.bus.loc[all_dcline["to_bus_id"], "lat"].values
+    all_dcline = project_branch(all_dcline)
+
+    dcline = all_dcline.iloc[
         9:,
```

Comment:
This will be okay for the current `"usa_tamu", "USA"` grid, but we will need to refactor if we ever pass a grid with a different set of DC line indices that are actually B2Bs.

Reply:
Do you want to use the dictionary you use in `power_flow_snapshot`?

Reply:
`plot_powerflow_snapshot` current takes a user input for which ones are DC lines, because it can be used for any interconnection. For this, we could let the user decide, or define a dict somewhere that if the `grid_model` is `usa_tamu`, then the b2b indices are `list(range(9))`. Either way.

Reply:
Done